File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        c_max = 0
        ans = nums[0]

        for num in nums:
            c_max = c_max + num
            ans = max(ans , c_max)
            if c_max <0:
                c_max=0
        
        return ans
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums)
    {
        unordered_set<int>s(nums.begin() , nums.end());
        int ans =0;
        for(int num : s)
        {
            // start from the start of the sequence
            if(s.find(num-1)==s.end())
            {
                int currentNum = num;

                int temp=1;

                while(s.find(currentNum+1)!=s.end())
                {
                    temp++;
                    currentNum++;

                }
                ans = max(ans , temp);
            }
        }
        return ans;
        
    }
};
